app/validators/url_validator.rb

```diff
@@ -1,7 +1,7 @@
 class UrlValidator < ActiveModel::EachValidator
   def validate_each(record, attribute, value)
-    unless URI.parse(value).kind_of?(URI::HTTP)
-      record.errors[attribute] << (options[:message] || "is an invalid URL")
-    end   
+    raise URI::InvalidURIError unless URI.parse(value).kind_of?(URI::HTTP)
+  rescue URI::InvalidURIError
+    record.errors[attribute] << (options[:message] || "is an invalid URL")
```

Comment:
Raising an exception and then immediately rescuing it seems bad. What if you separated the URL parsing from the bit dealing with the validation error?

```
unless valid_url?(value)
  record.errors[attribute] << (options[:message] || "is an invalid URL")
end

def valid_url?(value)
  URI.parse(value).kind_of?(URI::HTTP)
rescue URI::InvalidURIError
  false
end
```